iron-list.html

```diff
@@ -639,21 +638,16 @@
       if (this._lastVisibleIndexVal === null) {
         if (this.grid) {
           var lastIndex = this.firstVisibleIndex + this._estRowsInView * this._itemsPerRow - 1;
-          this._lastVisibleIndexVal = lastIndex > this._virtualCount ? this._virtualCount : lastIndex;
+          this._lastVisibleIndexVal = Math.min(this._virtualCount, lastIndex);
         } else {
           var physicalOffset = this._physicalTop;
-
           this._iterateItems(function(pidx, vidx) {
-            physicalOffset += this._getPhysicalSizeIncrement(pidx);
-
-            if(physicalOffset <= this._scrollBottom) {
-              if (this.grid) {
-                var lastIndex = vidx - vidx % this._itemsPerRow + this._itemsPerRow - 1;
-                this._lastVisibleIndexVal = lastIndex > this._virtualCount ? this._virtualCount : lastIndex;
-              } else {
-                this._lastVisibleIndexVal = vidx;
-              }
+            if (physicalOffset < this._scrollBottom) {
+              this._lastVisibleIndexVal = vidx;
+            } else {
+              return false;

```

Comment:
I was thinking:

```javascript
this._lastVisibleIndexVal = this._iterateItems(function(pidx, vidx) {
  physicalOffset += this._getPhysicalSizeIncrement(pidx);

  if (physicalOffset > this._scrollBottom) {
    return vidx;
  }
});
```

Not sure if this is entirely correct though. Also, note you have to return a truthy value to break out of `_iterateItems`.